impeller/golden_tests/metal_screenshoter.mm

```diff
@@ -0,0 +1,55 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/impeller/golden_tests/metal_screenshoter.h"
+
+#include <CoreImage/CoreImage.h>
+#include "impeller/renderer/backend/metal/context_mtl.h"
+#include "impeller/renderer/backend/metal/texture_mtl.h"
+#define GLFW_INCLUDE_NONE
+#include "third_party/glfw/include/GLFW/glfw3.h"
+
+namespace impeller {
+namespace testing {
+
+MetalScreenshoter::MetalScreenshoter() {
+  FML_CHECK(::glfwInit() == GLFW_TRUE);
+  playground_ = PlaygroundImpl::Create(PlaygroundBackend::kMetal);
+  aiks_context_.reset(new AiksContext(playground_->GetContext()));
+}
+
+std::unique_ptr<MetalScreenshot> MetalScreenshoter::MakeScreenshot(
+    Picture&& picture,
+    const ISize& size) {
+  std::shared_ptr<Image> image = picture.ToImage(*aiks_context_, size);
+  std::shared_ptr<Texture> texture = image->GetTexture();
+  id<MTLTexture> metal_texture =
+      std::static_pointer_cast<TextureMTL>(texture)->GetMTLTexture();
+
+  if (metal_texture.pixelFormat != MTLPixelFormatBGRA8Unorm) {
+    return {};
+  }
+
+  CIImage* ciImage = [[CIImage alloc] initWithMTLTexture:metal_texture
+                                                 options:@{}];
```

Comment:
This means this will only work with `kHostVisible` textures. That's probably fine for now, but should be documented.

If you add a blit pass here you can make it work for device private textures as well.

Reply:
I'm pretty sure that CoreImage is drawing the image to a host readable texture.  I tried to get the pixels directly from the metal texture but was unable to, so CoreImage was necessary.